R/summarize_qc_flags.R

```diff
@@ -519,52 +510,44 @@ get_custom_flags <- function(directory = here::here(),
       R_flag <- NA
       P_flag <- NA
       Cell_count <- NA
-      percent_missing <- NA
       RRU <- NA
 
       flags <- data.frame(
-        filename, column, A_flag, AE_flag, R_flag, P_flag,
-        Cell_count, percent_missing, RRU
+        filename = names(dfList)[i],
+        column,
+        Cell_count,
+        A_flag,
+        AE_flag,
+        R_flag,
+        P_flag,
+        RRU
       )
 
       cust_flags <- rbind(cust_flags, flags)
     }
   }
-  colnames(cust_flags)[7] <- "Data Totals"
 
   #generate summary statistics for each column:
   data_file_summaries <- cust_flags %>%
     dplyr::group_by(filename) %>%
-    dplyr::summarize(A_total = sum(A_flag),
-                     AE_total = sum(AE_flag),
-                     P_total = sum(P_flag),
-                     R_total = sum(R_flag),
-                     missing_mean_percent = mean(percent_missing),
-                     RRU_mean = mean(RRU),
-                     RRU_sd = stats::sd(RRU))
-
-  #generate data package level summaries
-  data_package_summary <- cust_flags %>%
-    plyr::summarize(A_total = sum(A_flag, na.rm = TRUE),
-                    AE_total = sum(AE_flag, na.rm = TRUE),
-                    P_total = sum(P_flag, na.rm = TRUE),
-                    R_total = sum(R_flag, na.rm = TRUE),
-                    missing_mean_percent = mean(percent_missing, na.rm = TRUE),
-                    RRU_mean = mean(RRU, na.rm = TRUE),
-                    RRU_sd = sd(RRU, na.rm = TRUE))
+    dplyr::summarize("A" = sum(A_flag),
+                     "AE" = sum(AE_flag),
+                     "P" = sum(P_flag),
+                     "R" = sum(R_flag),
+                     "% Accepted" = mean(RRU)) %>%
+    rename("File Name" = filename) %>%
+    mutate(`% Accepted` = scales::percent(`% Accepted`, accuracy = 0.1))
+
+  cust_flags <- cust_flags %>% mutate(column = str_remove(column, "_flag"), RRU = scales::percent(RRU, accuracy = 0.1)) %>% select("File Name" = filename, "Measure" = column, "Number of Records" = Cell_count, "A" = A_flag, "AE" = AE_flag, "R" = R_flag, "P" = P_flag, "% Accepted" = RRU)
+
 
   qc_summary <- list(cust_flags,
-                     data_file_summaries,
-                     data_package_summary)
+                     data_file_summaries)
 
   names(qc_summary) <- c("Column Level QC Summaries",
-                       "Data File Level QC Summaries",
-                       "Data Package Level QC Summaries")
+                         "Data Package Level QC Summaries")
 
   if (output == "package") {
-    return(qc_summary[[3]])
-  }
-  if (output == "files") {
```

Comment:
If you remove the output == files, what happens when someone calls the function using summarize_qc_flags(output = "files")?